sdk/core/azure-core-opentelemetry/test/ut/azure_core_otel_test.cpp

```diff
@@ -0,0 +1,690 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#define USE_MEMORY_EXPORTER 1
+#include "azure/core-opentelemetry/opentelemetry.hpp"
+#include <azure/core/test/test_base.hpp>
+#if defined(_MSC_VER)
+// The OpenTelemetry headers generate a couple of warnings on MSVC in the OTel 1.2 package, suppress
+// the warnings across the includes.
+#pragma warning(push)
+#pragma warning(disable : 4100)
+#pragma warning(disable : 4244)
+#endif
+#include <opentelemetry/exporters/memory/in_memory_span_data.h>
+#include <opentelemetry/exporters/memory/in_memory_span_exporter.h>
+#include <opentelemetry/exporters/ostream/span_exporter.h>
+#include <opentelemetry/sdk/common/global_log_handler.h>
+#include <opentelemetry/sdk/trace/exporter.h>
+#include <opentelemetry/sdk/trace/processor.h>
+#include <opentelemetry/sdk/trace/simple_processor.h>
+#include <opentelemetry/sdk/trace/tracer_provider.h>
+#if defined(_MSC_VER)
+#pragma warning(pop)
+#endif
+
+#include <chrono>
+#include <gtest/gtest.h>
+
+class OpenTelemetryTests : public Azure::Core::Test::TestBase {
+private:
+  class CustomLogHandler : public opentelemetry::sdk::common::internal_log::LogHandler {
```

Comment:
should this class definition be pulled out ?

Reply:
Why? I don't think it should be global, why do you think it should be global?

Reply:
nested private classes seems somehow dirty, but given that it's for testing purposes ... fine